src/radical/pilot/db/database.py

```diff
@@ -259,12 +259,71 @@ def pilot_command(self, cmd, arg=None, pids=None):
             raise RuntimeError ('pymongo error: %s' % e.details) from e
 
 
+    # --------------------------------------------------------------------------
+    #
+    def pilot_rpc(self, pid, rpc, args=None):
```

Comment:
Can we add a test for this method?

Reply:
Should that be an integration test then?    The logic and mechanics of the method is all around pushing something to MongoDB and waiting for a reply to pop up.  If MongoDB is mocked away, there is not much to test.  Should I convert some RPC example to a test, would that work in your opinion?

Reply:
Makes sense IMHO.

Reply:
That test has been added.